### PR TITLE
feat(RDS): add a datasource to get storageTypes

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
 make testacc TEST='./huaweicloud/services/acceptance/rds' TESTARGS='-run=TestAccDatasourceStorage'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds -v -run=TestAccDatasourceStorage -timeout 360m -parallel 4
=== RUN   TestAccDatasourceStoragetype_basic
=== PAUSE TestAccDatasourceStoragetype_basic
=== CONT  TestAccDatasourceStoragetype_basic
--- PASS: TestAccDatasourceStoragetype_basic (8.93s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rds       8.983s
```

### CODE REVIEW

File: docs/data-sources/rds_storage_types.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the data source.
+  If omitted, the provider-level region will be used.
+
+* `database_name` - (Required, String) DB engine. The valid values are **MySQL**, **PostgreSQL**, **SQLServer**.
```
Comment (Lance52259) [inline docs/data-sources/rds_storage_types.md]: `database_engine`?

Comment (hmbig2) [reply docs/data-sources/rds_storage_types.md]: ok

File: docs/data-sources/rds_storage_types.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  The options are as follows:
+    - **ULTRAHIGH**: SSD storage.
+    - **LOCALSSD**: Local SSD storage.
+    - **CLOUDSSD**: Cloud SSD storage.
+        This storage type is supported only with general-purpose and dedicated DB instances.
+    - **ESSD**: extreme SSD storage.
+        This storage type is supported only with dedicated DB instances.
+
+* `az_status` - Key indicates the AZ ID, and value indicates the specification status in the AZ.
+  The options of value are as follows:
```
Comment (Lance52259) [inline docs/data-sources/rds_storage_types.md]:
```suggestion
* `az_status` - The status details of the AZs to which the specification belongs.  
  The key indicates the AZ ID, and value indicates the specification status in the AZ.
  The options of value are as follows:
```

Comment (hmbig2) [reply docs/data-sources/rds_storage_types.md]: OK，done

File: docs/data-sources/rds_storage_types.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the data source.
+  If omitted, the provider-level region will be used.
+
+* `db_type` - (Required, String) DB engine. The valid values are **MySQL**, **PostgreSQL**, **SQLServer**.
+
+* `version_name` - (Optional, String) DB version number.
+
+* `ha_mode` - (Optional, String) HA mode. The valid values are **single**, **ha**, **replica**.
```
Comment (ShiChangkuo) [inline docs/data-sources/rds_storage_types.md]: it's better to keep the same parameter names as `data.huaweicloud_rds_flavors`

Comment (hmbig2) [reply docs/data-sources/rds_storage_types.md]: ok, done

**Pull request merged**